src/EnergyPlus/DataEnvironment.hh

```diff
@@ -144,7 +143,7 @@ struct EnvironmentData : BaseGlobalStruct
     Real64 WaterMainsTemp = 0.0;                                // Current water mains temperature
     int Year = 0;                                               // Current calendar year of the simulation from the weather file
     int YearTomorrow = 0;                                       // Tomorrow's calendar year of the simulation
-    Array1D<Real64> SOLCOS = Array1D<Real64>(3);                // Solar direction cosines at current time step
+    Vector3<Real64> SOLCOS = {0.0, 0.0, 0.0};                                     // Solar direction cosines at current time step
```

Comment:
A `Vector3` is not a dynamic array, in fact it isn't an array at all, it's a `struct` that also happens to define `(1)`, `(2)`, and `(3)` as operators as well as other geometric operators. For geometric points or vectors, always use `Vector3` rather than an array of size 3.